src/d/actor/d_a_shutter.cpp

```diff
@@ -5,80 +5,292 @@
+    if (mCount >= 0) {
+        mCount--;
+    }
+    demo();
+    set_mtx();
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mdBgW[i]->Move();
+    }
+    mSwitch = fopAcM_isSwitch(this, mNo);
+    return TRUE;
 }
 
+const char* M_act_table[4] = {"WAIT", "WAIT02", "OPEN", "CLOSE"};
+
 /* 00000788-00000B14       .text shutter_move__11daShutter_cFv */
 void daShutter_c::shutter_move() {
-    /* Nonmatching */
+    int actionIndex = dComIfGp_evmng_getMyActIdx(mStaffId, (char**)M_act_table, ARRAY_SIZE(M_act_table), FALSE, 0);
+
+    float maxVel;
+    float minVel;
+
+    maxVel = m_max_speed[mType];
+    minVel = m_min_speed[mType];
+
+    float fVar4;
+    float fVar5;
+
+    switch (actionIndex) {
+        case 0: //WAIT
+        {
+            mTimer = 0xf;
+            field_0x33A = 0;
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+        case 1: //WAIT02
+        {
+            if (!cLib_calcTimer((u8*)&mTimer)) {
+                field_0x339 = 0;
+                dComIfGp_evmng_cutEnd(mStaffId);
+            }
+            return;
+        }
+        case 2: //OPEN
+        {
+            field_0x33A++;
+            if (field_0x339 == 0) {
+                field_0x339 = 1;
+                if (dStage_stagInfo_GetSaveTbl(dComIfGp_getStage().getStagInfo()) == 5) {
+                    fopAcM_seStart(this, JA_SE_OBJ_B_SHUTTER_OPEN, 0);
+                }
+            }
+            fVar4 = cLib_addCalc(&mcXyz[0].x, -m_move_len[mType] - m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            fVar5 = cLib_addCalc(&mcXyz[1].x, m_move_len[mType] + m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            if ((dStage_stagInfo_GetSaveTbl(dComIfGp_getStage().getStagInfo()) == 5) && (field_0x33A == 75)) {
+                fopAcM_seStart(this, JA_SE_OBJ_B_SHUTTER_STOP, 0);
+                dComIfGp_getVibration().StartShock(4, -0x21, cXyz(0.0f, 1.0f, 0.0f));
+            }
+            if (fVar4 != 0.0f){
+                return;
+            } 
+            if (fVar5 != 0.0f) {
+                return;
+            }
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+        case 3: //CLOSE
+        {
+            if (field_0x339 == 0) {
+                field_0x339 = 1;
+            }
+            fVar4 = cLib_addCalc(&mcXyz[0].x, -m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            fVar5 = cLib_addCalc(&mcXyz[1].x, m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            if (fVar4 != 0.0f) {
+                return;
+            }
+            if (fVar5 != 0.0f) {
+                return;
+            }
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+    }
+    dComIfGp_evmng_cutEnd(mStaffId);
 }
 
 /* 00000B14-00000CF0       .text demo__11daShutter_cFv */
 void daShutter_c::demo() {
-    /* Nonmatching */
+    u8 bVar3 = fopAcM_isSwitch(this, mNo);
+    if (field_0x320 == 0) {
+        if ((bVar3 != (u8)mSwitch) && (mCount < 0)) {
```

Comment:
```suggestion
    bool isSwitch = fopAcM_isSwitch(this, mSwitchNo);
    if (field_0x320 == 0) {
        if ((isSwitch != mbIsSwitch) && (mCount < 0)) {
```
Does this still match if you change mbIsSwitch from s8 -> bool and the local variable from u8 -> bool as well? There are some cases where you might need to use `u8` for a boolean to get it to match, but most of the time `bool` or `BOOL` would be correct.

Reply:
No, I couldn't get it to match when setting both to either `bool` or `BOOL`

Reply:
Interesting, does it match with mbIsSwitch changed from s8 -> u8? (Instead of casting it to u8)

Reply:
Yep. Just committed that change